### ISSUE TITLE
Trying to open WiFi settings in iOS 18 opens the general settings page instead

### ISSUE THREAD

**Issue by adamk22** [yanncabral/open_settings_plus#23]
Seems this plugin is having issues with iOS 18 (possibly related #20 and #21 ) since calling the WiFi method does not open the wifi settings but opens up the general settings page instead.

**Comment by PBomermustekiin** [comment]
same for photos and camera method. We are waiting an update...

**Comment by EnzoPortela** [comment]
Same here

**Comment by yanncabral** [comment]
Hello! I'm glad to hear that the library is helping you.

This is a common issue, as the path string used to open a specific page may change across different iOS versions. Simply updating the string to match the new pattern would cause it to stop working on older versions, and vice versa. 

To definitively resolve this, we need to map the paths for various iOS (and also Android) versions and use the correct path based on the detected version. I'm currently working on this and it will be addressed soon.

The library already supports opening customizable paths, so it's possible to implement this check directly within your application. I'm developing a solution that covers all relevant versions, but I need to complete another task I'm currently handling first.

If you'd like, feel free to open a pull request if you can't wait a bit. Either way, an update will be released within a few days.


Thank you for your understanding and support!

**Comment by PBomermustekiin** [comment]
Hello! Do you have any updates? We are eagerly waiting together and excited.

**Comment by PBomermustekiin** [comment]
@yanncabral  Hi, the issue in iOS 18 still continue after your new update.

**Comment by ivanhercaz** [comment]
Hi @yanncabral!

I researched a bit to know if I could help you with this issue, at least for the iOS side, but I feel a bit confused because I didn't find any official resource about the paths used by iOS across versions. So I asked in the Apple Developer Forums, I leave you the link here in case you want to join the question: https://forums.developer.apple.com/forums/thread/770783

The most non-official up-to-date list I found was the one by Wesley de Groot, see [iOS Settings URL's](https://wesleydegroot.nl/blog/iOS-settings-URLs).

In any case, it seems an easy issue that it is harder than the expected because of the lack of documentation and procedures related with the topic. So take your time without stressing yourself! ❤️ If I discover something more, I will share with you.

**Comment by PBomermustekiin** [comment]
Hello @yanncabral, is there any update?

**Comment by zherman0981** [comment]
I don't typically comment on issues but this one was driving me crazy! For anyone who is still running into this issue, it appears that Apple has deprecated the use of these URLs so it doesn't seem possible to utilize this package anymore, at least for newer os versions. I also found places where it mentions the app will be blocked from the app store if such urls exist.. 

To get around this, I found that using the [Geolocator](https://pub.dev/packages/geolocator) package allows programmatic navigation to the apps settings which will allow the user to enable location services from there. Works for both iOS and Android so thought I would mention it here. Hope this helps!

**Comment by zherman0981** [comment]
> > I don't typically comment on issues but this one was driving me crazy! For anyone who is still running into this issue, it appears that Apple has deprecated the use of these URLs so it doesn't seem possible to utilize this package anymore, at least for newer os versions. I also found places where it mentions the app will be blocked from the app store if such urls exist..
> > To get around this, I found that using the [Geolocator](https://pub.dev/packages/geolocator) package allows programmatic navigation to the apps settings which will allow the user to enable location services and other permissions from there. Works for both iOS and Android so thought I would mention it here. Hope this helps!
> 
> For Location Services on iOS 18+, The [Geolocator](https://pub.dev/packages/geolocator) package only displays a dialog with instructional text, <img alt="Image" width="260" src="https://private-user-images.githubusercontent.com/97861347/421051849-3e457d30-3a5f-403d-80e8-96646330cd03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MzQxMDIsIm5iZiI6MTc0MTYzMzgwMiwicGF0aCI6Ii85Nzg2MTM0Ny80MjEwNTE4NDktM2U0NTdkMzAtM2E1Zi00MDNkLTgwZTgtOTY2NDYzMzBjZDAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE5MTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmYWFkMDk0MjBkYzExNDliMDdhZDgyZTNiMGFjODIwOWM5ZDBjOWY2NWZjY2YwY2RlZjk2OWY5OTE1NDgxMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3kWFR_I34_ISsPpJl8H9yZ2m3pJnryrWy1X3fyqCXeU"> whereas [Google Maps Flutter](https://pub.dev/packages/google_maps_flutter) can still prompt users directly to enable Location Services (albeit only twice) <img alt="Image" width="219" src="https://private-user-images.githubusercontent.com/97861347/421052500-6d52bd68-f973-435d-be69-e0f77e67cf1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MzQxMDIsIm5iZiI6MTc0MTYzMzgwMiwicGF0aCI6Ii85Nzg2MTM0Ny80MjEwNTI1MDAtNmQ1MmJkNjgtZjk3My00MzVkLWJlNjktZTBmNzdlNjdjZjFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE5MTAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4Y2Y2NDlkNjNjOTIwYmUzM2JiNmEyODI4NTg5ZjVmZTI1YTAwMTBlMzYyMTU2NjEzNWM5YmQ4ODIwM2VmOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FtkmO6ct61Pd-gUk4UaTt9DwzFYWQjfDa1Y73G78e58">

That is true for your above case if location is disabled entirely. After its enabled, it will navigate to the apps settings. Did you use `Geolocator.appSettings()` method or the one for location services?

**Comment by bruno-m-antunes-alb** [comment]
> Hi [@yanncabral](https://github.com/yanncabral)!
> 
> I researched a bit to know if I could help you with this issue, at least for the iOS side, but I feel a bit confused because I didn't find any official resource about the paths used by iOS across versions. So I asked in the Apple Developer Forums, I leave you the link here in case you want to join the question: https://forums.developer.apple.com/forums/thread/770783
> 
> The most non-official up-to-date list I found was the one by Wesley de Groot, see [iOS Settings URL's](https://wesleydegroot.nl/blog/iOS-settings-URLs).
> 
> In any case, it seems an easy issue that it is harder than the expected because of the lack of documentation and procedures related with the topic. So take your time without stressing yourself! ❤️ If I discover something more, I will share with you.

@ivanhercaz did you manage to use the "paths" on that Wesley's list using the "sendCustomMessage" ?

I'm trying on iOS 26.3 without success.

**Comment by ivanhercaz** [comment]
> > Hi [@yanncabral](https://github.com/yanncabral)!
> > I researched a bit to know if I could help you with this issue, at least for the iOS side, but I feel a bit confused because I didn't find any official resource about the paths used by iOS across versions. So I asked in the Apple Developer Forums, I leave you the link here in case you want to join the question: https://forums.developer.apple.com/forums/thread/770783
> > The most non-official up-to-date list I found was the one by Wesley de Groot, see [iOS Settings URL's](https://wesleydegroot.nl/blog/iOS-settings-URLs).
> > In any case, it seems an easy issue that it is harder than the expected because of the lack of documentation and procedures related with the topic. So take your time without stressing yourself! ❤️ If I discover something more, I will share with you.
> 
> [@ivanhercaz](https://github.com/ivanhercaz) did you manage to use the "paths" on that Wesley's list using the "sendCustomMessage" ?
> 
> I'm trying on iOS 26.3 without success.

Hi @bruno-m-antunes-alb!

If I am not remembering wrong, yes, I managed to use them but I am not sure for which version. I just reviewed the list of Wesley (@0xWDG), and I just see his list is updated for iOS 26.1. I will try to test it this weekend, if I discover something I will share it with you.
